File: web-smartstock/src/app/dtos/product.ts
export class ProductDto {
    constructor(
        public category: string,
        public created_at: Date,
        public currency_price: string,
        public description: string,
        public discount_price: number,
        public expired_at: Date,
        public id: string,
        public is_promotion: boolean,
        public manufacturer_id: string,
        public name: string,
        public store_conditions: string,
        public unit_price: number,
        public updated_at: Date,
        public url_photo: string
    ) {}
}

export interface PaginatedProducts {
    data: ProductDto[];
    page: number;
    per_page: number;
    total: number;
    total_pages: number;
}

File: mobile-smartstock/app/src/main/java/com/smartstock/myapplication/models/Product.kt
package com.smartstock.myapplication.models



data class Product(
    var id: String,
    val manufacturer_id: String,
    val name: String,
    val description: String,
    val category: String,
    val unit_price: Double,
    val currency_price: String,
    val is_promotion: Boolean,
    val discount_price: Double,
    val expired_at: String,
    val url_photo: String,
    val store_conditions: String,
    var quantity: Int? = null
)